python/lsst/ctrl/bps/wms/htcondor/htcondor_service.py

```diff
@@ -358,31 +363,20 @@ def from_generic_workflow(cls, config, generic_workflow, out_prefix, service_cla
                                       "bps_run_quanta": create_count_summary(generic_workflow.quanta_counts),
                                       "bps_job_summary": create_count_summary(generic_workflow.job_counts)})
 
-        # Determine the hard limit for the memory requirement.
-        found, limit = config.search('memoryLimit')
-        if not found:
-            search_opts = {"default": DEFAULT_HTC_EXEC_PATT}
-            _, site = config.search("computeSite")
-            if site:
-                search_opts["curvals"] = {"curr_site": site}
-            _, patt = config.search("executeMachinesPattern", opt=search_opts)
-
-            # To reduce the amount of data, ignore dynamic slots (if any) as,
-            # by definition, they cannot have more memory than
-            # the partitionable slot they are the part of.
-            constraint = f'SlotType != "Dynamic" && regexp("{patt}", Machine)'
-            pool_info = condor_status(constraint=constraint)
-            try:
-                limit = max(int(info["TotalSlotMemory"]) for info in pool_info.values())
-            except ValueError:
-                _LOG.debug("No execute machine in the pool matches %s", patt)
-        if limit:
-            config[".bps_defined.memory_limit"] = limit
+        site_values = {}  # cache compute site specific values to reduce config lookups
```

Comment:
Can we move this line closer to the place where it is used (e.g. put it either before or after the line 373)?